crates/sui-proxy/src/consumer.rs

```diff
@@ -236,21 +236,62 @@ async fn check_response(
     }
 }
 
+async fn convert(
+    mfs: Vec<MetricFamily>,
+) -> Result<impl Iterator<Item = WriteRequest>, (StatusCode, &'static str)> {
+    let result = match tokio::task::spawn_blocking(|| {
+        let timer = CONSUMER_OPERATION_DURATION
+            .with_label_values(&["convert_to_remote_write_task"])
+            .start_timer();
+        let result = Mimir::from(mfs);
+        timer.observe_duration();
+        result.into_iter()
+    })
+    .await
+    {
+        Ok(v) => v,
+        Err(err) => {
```

Comment:
Could probably just be a `map_err`?

Reply:
haha, it can, but not this round.  i have to refactor too much of the handlers to make that work.